### DISCUSSION TITLE
스레드 캐시 관련

### DISCUSSION THREAD

**Discussion by emost22** [Well-Founded-Confidence/RealMySQL#20] [Q&A]
책 내용 중에
- 클라이언트가 커넥션을 종료하면 해당 스레드는 다시 스레드 캐시로 돌아간다.
- 이미 스레드 캐시에 일정 갯수 이상의 스레드가 대기 중이면 해당 스레드를 종료한다.

라는 내용이 있습니다.

서버 실행 시에 thread_cache_size 만큼의 스레드가 이미 생성되고, 사용 중인 스레드는 캐시 내에 남아 있는걸로 알고 있는데
일정 갯수 이상의 스레드가 대기 중일 수는 없지 않나요?
아니면 일정 갯수보다 많은 스레드가 캐시 내에서 생성되거나 존재할 수 있는 건가요?

**Comment by menthamin** [comment]
스레드 풀에서 스레드 캐시가 사용된다.
- 스레드 풀: CPU 사용률을 줄이는 목적 (효율적으로 스레드를 사용해서)
- 스레드 캐시: 재사용

---
**스레드 cache_size를 초과했을 때 추가 스레드를 생성?**

Auto Scale?

캐시에 둔다.
- 나중에 사용하기 위해서 임시로 둔다.
- 더이상 내놓을 스레드가 없을 때 

캐시에 스레드가 왜 들어가는지?
- 스레드 생성에 오버헤드
- MVCC 버전 체킹
